packages/embed-components/src/Theme/state/sagas.ts

```diff
@@ -0,0 +1,136 @@
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+import { takeEvery, put, call } from 'typed-redux-saga'
+import type { PayloadAction } from '@reduxjs/toolkit'
+import type { ITheme } from '@looker/sdk'
+import {
+  registerThemeService,
+  getThemeService,
+  EmbedUrl,
+} from '@looker/embed-services'
+import { themeActions } from './slice'
+import type { SelectThemeAction } from './slice'
+
+/**
+ * Registers theme service with factory
+ */
+function* initSaga() {
+  const { initSuccessAction, setFailureAction } = themeActions
+  try {
+    registerThemeService(10)
+    yield* put(initSuccessAction())
+  } catch (error: any) {
+    yield* put(setFailureAction({ error: error.message }))
+  }
+}
+
+/**
+ * Gets all themes, default and url themes, and sets the selected theme
+ */
+function* loadThemeDataSaga() {
+  const { loadThemeDataSuccessAction, setFailureAction } = themeActions
+  try {
+    const service = getThemeService()
+    yield* call([service, 'load'])
+    const searchParams = new EmbedUrl().searchParams
+    const urlThemeName = searchParams.theme
+    let urlTheme: ITheme | undefined
+    if (urlThemeName) {
+      urlTheme = service.find('name', urlThemeName)
+    }
+    yield* put(
+      loadThemeDataSuccessAction({
+        themes: service.items,
+        defaultTheme: service.defaultTheme!,
+        selectedTheme: (urlTheme ?? service.defaultTheme)!,
+      })
+    )
+  } catch (error: any) {
+    yield* put(setFailureAction({ error: error.message }))
+  }
+}
+
+/**
+ * Gets all available themes
+ */
+function* getThemesSaga() {
+  const { getThemesSuccessAction, setFailureAction } = themeActions
+  try {
+    const service = getThemeService()
+    yield* call([service, 'getAll'])
+    yield* put(getThemesSuccessAction({ themes: service.items }))
+  } catch (error: any) {
+    yield* put(setFailureAction({ error: error.message }))
+  }
+}
+
+/**
+ * Gets default theme
+ */
+function* getDefaultThemeSaga() {
+  const { getDefaultThemeSuccessAction, setFailureAction } = themeActions
+  try {
+    const service = getThemeService()
+    const defaultTheme = yield* call([service, 'getDefaultTheme'])
+    yield* put(getDefaultThemeSuccessAction({ defaultTheme }))
+  } catch (error: any) {
+    yield* put(setFailureAction({ error: error.message }))
+  }
+}
+
+/**
+ * Sets the selected theme by id
+ * @param action containing id of theme to select
+ */
+function* selectThemeSaga(action: PayloadAction<SelectThemeAction>) {
+  const { selectThemeSuccessAction, setFailureAction } = themeActions
+  try {
+    const service = getThemeService()
+    if (service.expired()) {
+      yield* call(getThemesSaga)
+      yield* call(getDefaultThemeSaga)
+    }
```

Comment:
@bryans99 are you ok with me reusing the other two sagas from this saga?

Reply:
Why not put the action that triggers it. This will block

Reply:
sure, I could do that

Reply:
I changed it to call the actions (not put, as I want it to block).